internal/security/secretstoreclient/mocks/mock.go

```diff
@@ -92,3 +92,15 @@ func (m *MockSecretStoreClient) RegenRootToken(vmkReader io.Reader, rootToken *s
 	arguments := m.Called(vmkReader, rootToken)
 	return arguments.Error(0)
 }
+
+func (m *MockSecretStoreClient) CheckSecretEngineInstalled(token string, mountPoint string, engine string) (isInstalled bool, err error) {
```

Comment:
Should we have a test case where if kv engine not enabled but we attempt a secret store it fails?

Reply:
Impossible to do with a unit test.  That would actually require changing the code so that it DIDN'T enable the KV engine, then actually spinning up a real Vault instance, then running a test to see if a secret couldn't be stored.  That is all impossible with "go test".    "go test" must test business logic without doing any I/O to actual instances of Vault.